Crud.php

```diff
@@ -0,0 +1,271 @@
+
+            // build the query for the doctrine paginator
+            $query = $numbers->createQueryBuilder('n');
+
+            //setting sorting
+            $query = $query->orderBy('n.slug', $sorting);
+
+            //Adding searching parametters
+            if(!empty($filters['slug'])) {
+                $query = $query->andWhere('n.slug like :slug');
+            }
+            if(!empty($filters['title'])) {
+                $query = $query->andWhere('n.title like :title');
+            }
+            if(!empty($filters['number'])) {
+                $query = $query->andWhere('n.number like :number');
+            }
+            if(!empty($filters['text'])) {
+                $query = $query->andWhere('n.text like :text');
+            }
+            if(!empty($filters['transcription'])) {
+                $query = $query->andWhere('n.transcription like :transcription');
+            }
+
+            //setting parametters for query
+            if(!empty($filters['slug'])) {
+                $query = $query->setParameter('slug', $filters['slug']);
+            } if(!empty($filters['title'])) {
+                $query = $query->setParameter('title', $filters['title']);
+            } if(!empty($filters['number'])) {
+                $query = $query->setParameter('number', $filters['number']);
+            } if(!empty($filters['text'])) {
+                $query = $query->setParameter('text', $filters['text']);
+            } if(!empty($filters['transcription'])) {
+                $query = $query->setParameter('transcription', $filters['transcription']);
+            }
+
+            $query = $query->getQuery();
+
+            //set page size
+            $pageSize = self::PER_PAGE;
+
+            // load doctrine Paginator
+            $paginator = new \Doctrine\ORM\Tools\Pagination\Paginator($query);
+
+            // you can get total items
+            $totalItems = count($paginator);
+
+            // get total pages
+            $pagesCount = ceil($totalItems / $pageSize);
+
+            // now get one page's items:
+            $paginator
+                ->getQuery()
+                ->setFirstResult($pageSize * ($page-1)) // set the offset
+                ->setMaxResults($pageSize); // set the limit
+
+            $result = [
+                'page' => $page,
+                'data' => $paginator,
+                'pages' => $pagesCount
+            ];
+
+            return $result;
+        }
+
+        /**
+         * Update number
+         * 
+         * @param int id
+         * @param string slug 
+         * @param string title
+         * @param int number 
+         * @param string text 
+         * @param string transcription
+         * 
+         * @return bool 
+         */
+        public function updateNumber(int $id, string $slug, string $title, int $number, string $text, string $transcription):bool {
+            $numberEntity = $this->entityManager->find('Number', $id);
```

Comment:
is it possible to extract updateNumber and createNumber  common functionality?

Reply:
Extracted to Entity